### PR TITLE
45 Fan Support Helper Tests

### LINKED ISSUE #45: Fan Support
Add support for controlling a fan using the Pi4J library. This involves creating a class that handles fan initialization, speed control, and state management.
**Tasks:**
    

1.  Include a circuit diagram and breadboard layout.
2. Integrate GPIO pin configuration for fan control. Like yaml configurations
3. Create FanHelper.java class.
4. Implement methods for:
Starting the fan
Stopping the fan
Setting the speed of the fan
5. Provide functioning code to integrate the component with Micronaut.

**Acceptance Criteria:** 

  1. FanController.java class with methods: start(), stop(), and setSpeed(int speed).
  2. GPIO pin configuration.
  3. Component operates correctly on hardware.

### PR DESCRIPTION
**Pull request by Ed0827**
Fixes #45

Updated the Gradle build configuration to include support for Pi4J Micronaut. Added the fan control feature with support for turning on, turning off, and setting the fan speed using PWM.

Created FanController.java to provide HTTP endpoints for fan control and FanHelper.java to handle the logic for fan operations, including starting, stopping, and adjusting speed.

Implemented methods: start() to start the fan at full speed, stop() to stop the fan, and setSpeed(int speed) to set the fan speed using PWM with a value range of 0 to 1024.

### CODE REVIEW

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
1) Delete the FanController that was defined under the inputdevices
2) Delete the Pi4J cofiguration from the yml file 
3) Change the class name for FanHelper under unittests
4) Please change the class name in FanHelper.java same as file name, followed by constructor too.

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/FanHelper.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.pi4j.io.pwm.Pwm;
+import com.pi4j.io.pwm.PwmConfig;
+import com.pi4j.io.pwm.PwmType;
+import com.pi4j.plugin.pigpio.provider.pwm.PiGpioPwmProviderImpl;
+import com.pi4j.library.pigpio.PiGpio;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FanHelper {
+
```
Comment (ThomasMacas) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/FanHelper.java:12]: Looks great, although this white space could do to be deleted as many of the other __Helper.java files have the final variable without whitespace after the function.

File: pi4micronaut-utils/src/main/resources/application.yml
Comment (kate-holdener) [inline pi4micronaut-utils/src/main/resources/application.yml:1]: What's this change for?

**Comment by kate-holdener** [comment]
Looks like FanController is missing (probably should be there under outputdevices?) The yml configuration for a fan is also missing.

**Review by yrlmanoharreddy — APPROVED** [review]
Added FanHelper.java class to manage fan initialization and speed control using the Pi4J library. Implemented methods to start, stop, and adjust the fan speed, along with an example controller demonstrating how to use FanHelper.java. Integrated GPIO pin configuration via YAML for seamless setup.

**Pull request merged**